### PR TITLE
[VIDEOPRT] Fix IRP_MN_FILTER_RESOURCE_REQUIREMENTS case

### PR DESCRIPTION
**Pull request by SergeGautherie**
## Purpose

Addendum to 25eae4b (r54237).
JIRA issue: [CORE-17651](https://jira.reactos.org/browse/CORE-17651) (Point 1)

## Proposed changes

Return requirements, or free them.

### CODE REVIEW

File: win32ss/drivers/videoprt/dispatch.c [hunk 1]
```diff
@@ -1029,7 +1029,11 @@ IntVideoPortDispatchFdoPnp(
             if (NT_SUCCESS(Status) && NT_SUCCESS(Irp->IoStatus.Status))
                 Status = IntVideoPortFilterResourceRequirements(DeviceObject, Irp);
             Irp->IoStatus.Status = Status;
-            Irp->IoStatus.Information = 0;
+            if (!NT_SUCCESS(Status) && Irp->IoStatus.Information != 0)
```
Comment (tkreuzer) [inline win32ss/drivers/videoprt/dispatch.c:1032]: This shouldn't be done here, but in whatever function has allocated it.

**Review by tkreuzer — CHANGES_REQUESTED** [review]
Please move cleanup to where the allocation happened.

File: win32ss/drivers/videoprt/dispatch.c (hunk 1 above)
Comment (SergeGautherie) [reply win32ss/drivers/videoprt/dispatch.c]: Moved.

File: win32ss/drivers/videoprt/resource.c [hunk 2]
```diff
@@ -93,7 +93,14 @@ IntVideoPortFilterResourceRequirements(
 
     Status = IntVideoPortGetLegacyResources(DriverExtension, DeviceExtension, &AccessRanges, &AccessRangeCount);
     if (!NT_SUCCESS(Status))
+    {
```
Comment (Extravert-ir) [inline win32ss/drivers/videoprt/resource.c:96]: I feel like we shouldn't touch `OldResList` at all on failure

**Review by Extravert-ir — CHANGES_REQUESTED** [review]

**Comment by Extravert-ir** [comment]
Fixed upstream

**Pull request closed without merging**